### PR TITLE
Also write out an .env.sh file which can be safely be 'source'd into a shell.

### PR DESCRIPTION
**Pull request by richardpark-msft**

### CODE REVIEW

File: tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1 [hunk 1]
```diff
@@ -12,22 +12,47 @@ if ([string]::IsNullOrEmpty($outputFile)) {
...
+        $envLines += $line
     }
 }
 
+$environmentText = $envLines -join "`n"
+
 $bytes = [System.Text.Encoding]::UTF8.GetBytes($environmentText)
 Set-Content $outputFile -Value $bytes -AsByteStream -Force
+
+function Write-Sourceable-BashFile([string]$sourceableBashFilePath) {
```
Comment (benbp) [inline tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1:38]: Missing the envlines function parameter. Powershell scopes are annoying so it's pulling in the script scoped var anyways.

Comment (richardpark-msft) [reply tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1:38]: Yeah, I wasn't sure if we "liked" that or not. I can also make the other one a function to reduce the scope.

File: tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1 [hunk 2]
```diff
@@ -12,22 +12,47 @@ if ([string]::IsNullOrEmpty($outputFile)) {
...
 foreach ($entry in $DeploymentOutputs.GetEnumerator()) {
     $line = "$($entry.name)=$($entry.value)"
     if (!$duplicates.Contains($line)) {
         $duplicates[$line] = $true
-        $environmentText += ("$line" + "`n")
+        $envLines += $line
     }
 }
 
+$environmentText = $envLines -join "`n"
```
Comment (benbp) [inline tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1:33]: I assume no trailing newline is going to break anything here but just putting in a note to remember.

File: tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1 (hunk 1 above)
Comment (richardpark-msft) [reply tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1:38]: Oh, I forgot I left the parameter in. Yeah, I'll bring it back, I kind of liked the scoping.

File: tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1 (hunk 2 above)
Comment (richardpark-msft) [reply tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1:34]: Fixed.

File: tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1 (hunk 1 above)
Comment (richardpark-msft) [reply tools/stress-cluster/cluster/kubernetes/stress-test-addons/images/test-resource-deployer/test-resources-post.ps1]: Bring it all back.

**Comment by richardpark-msft** [comment]
CC: @LarryOsterman

**Comment by richardpark-msft** [comment]
I just need to test this.

**Review by benbp — APPROVED** [review]

**Comment by weshaggard** [comment]
@richardpark-msft is this still interesting? Is there no way to avoid writing these secrets to disk?

**Comment by richardpark-msft** [comment]
> @richardpark-msft is this still interesting? Is there no way to avoid writing these secrets to disk?

They weren't necessarily secrets (for instance, one of the env variables is the endpoint you connect to, without any keys). However, the original reason the .env file wasn't source-able was because of connection strings, which have embedded '='s in them. Given our push away from those cred types, I think we can just drop this PR altogether!

**Pull request closed without merging**